src/søknad/steg/1-omdeg/medlemskap/Medlemskap.tsx

```diff
@@ -10,36 +11,83 @@ import { usePersonContext } from '../../../../context/PersonContext';
 import useSøknadContext from '../../../../context/SøknadContext';
 import PeriodeBoddIUtlandet from './PeriodeBoddIUtlandet';
 import SeksjonGruppe from '../../../../components/gruppe/SeksjonGruppe';
+import { IMedlemskap } from '../../../../models/omDeg';
+import { injectIntl, IntlShape } from 'react-intl';
 
-const Medlemskap: React.FC = () => {
+const Medlemskap: React.FC<{ intl: IntlShape }> = ({ intl }) => {
   const { person } = usePersonContext();
-  const { søknad } = useSøknadContext();
+  const { søknad, settSøknad } = useSøknadContext();
   const { statsborgerskap } = person.søker;
-  const { søkerBosattINorgeSisteTreÅr, søkerOppholderSegINorge } = søknad;
+  const {
+    søkerOppholderSegINorge,
+    søkerBosattINorgeSisteTreÅr,
+    søkerErFlyktning,
+  } = søknad.medlemskap;
+
+  const settMedlemskapBooleanFelt = (
+    spørsmål: IJaNeiSpørsmål,
+    svar: boolean
+  ) => {
+    settSøknad({
+      ...søknad,
+      medlemskap: {
+        ...søknad.medlemskap,
+        [spørsmål.spørsmål_id]: {
+          label: intl.formatMessage({ id: spørsmål.tekstid }),
+          verdi: svar,
+        },
+      },
+    });
+  };
+
+  const hentValgtSvar = (spørsmål: IJaNeiSpørsmål, medlemskap: IMedlemskap) => {
+    for (const [key, value] of Object.entries(medlemskap)) {
+      if (key === spørsmål.spørsmål_id && value !== undefined) {
+        return value.verdi;
+      }
+    }
+  };
 
   return (
     <SeksjonGruppe>
-      <KomponentGruppe>
-        <JaNeiSpørsmål spørsmål={oppholderSegINorge} />
+      <KomponentGruppe key={oppholderSegINorge.spørsmål_id}>
+        <JaNeiSpørsmål
+          spørsmål={oppholderSegINorge}
+          valgtSvar={hentValgtSvar(oppholderSegINorge, søknad.medlemskap)}
+          onChange={settMedlemskapBooleanFelt}
+        />
       </KomponentGruppe>
 
-      {typeof søkerOppholderSegINorge === 'boolean' ? (
-        <KomponentGruppe>
-          <JaNeiSpørsmål spørsmål={bosattINorgeDeSisteTreÅr} />
+      {typeof søkerOppholderSegINorge?.verdi === 'boolean' ? (
```

Comment:
Hvorfor gjør dere en sjekk på typen til spørsmålet?

Reply:
Var for å sjekke om søkerOppholderSegINorge var blitt satt (true eller false), og det ikke hold med å sjekke `søkerOppholderSegINorge ? ` siden denne kodelinjen sjekker om den eksisterer / er true. 

Men har funnet en litt mer forståelig erstatning.
`søkerOppholderSegINorge.hasProperty('verdi') ? (...) : null;`

Er seff åpen for andre forslag :)